src/database/hooks/useLocalDatabaseHook.js

```diff
@@ -10,14 +10,15 @@ import {InitialStartupAtom} from '../../service/initialStartup';
 
 const MIGRATION_STATUS = {
   NOT_MIGRATED: 'NOT_MIGRATED',
-  MIGRATED: 'MIGRATED'
+  MIGRATED: 'MIGRATED',
+  MIGRATING: 'MIGRATING'
 };
 
 /**
  *
  * @returns {UseLocalDatabaseHook}
  */
-const useLocalDatabaseHook = () => {
+const useLocalDatabaseHook = (withMigration = false) => {
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The introduction of the `withMigration` flag in the `useLocalDatabaseHook` function is a good approach. This allows the caller to decide whether they want to perform migrations or not, providing more flexibility.

```diff
- const useLocalDatabaseHook = () => {
+ const useLocalDatabaseHook = (withMigration = false) => {
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->